### PR TITLE
`azurerm_site_recovery_replication_recovery_plan`: limit set `replicated_protected_items` for `Shutdown` type `recovery_group`

### LINKED ISSUE #22258: azurerm_site_recovery_replication_recovery_plan resorts recovery_groups on every run although there are no changes
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.4

### AzureRM Provider Version

3.62.0

### Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replication_recovery_plan 

### Terraform Configuration Files

```hcl
resource "azurerm_site_recovery_replication_recovery_plan" "recovery_plan" {
  name                      = "example-recovery-plan"
  recovery_vault_id         = data.azurerm_recovery_services_vault.asr-vault.id
  source_recovery_fabric_id = data.azurerm_site_recovery_fabric.asr-recovery-fabric.id
  target_recovery_fabric_id = data.azurerm_site_recovery_fabric.asr-recovery-fabric.id

  recovery_group {
    type                       = "Shutdown"
    replicated_protected_items = local.all_replication_protected_items

    pre_action {
      name            = "some-pre-action"
      type            = "AutomationRunbookActionDetails"
      fabric_location = "Primary"

      fail_over_directions = [
        "PrimaryToRecovery", "RecoveryToPrimary"
      ]
      fail_over_types = [
        "TestFailover", "PlannedFailover", "UnplannedFailover"
      ]
      runbook_id = module.asr-dr-runbooks.automation-runbook-some-pre-action-id
    }

    pre_action {
      name            = "stop-applications"
      type            = "AutomationRunbookActionDetails"
      fabric_location = "Primary"

      fail_over_directions = [
        "PrimaryToRecovery", "RecoveryToPrimary"
      ]
      fail_over_types = [
        "TestFailover", "PlannedFailover", "UnplannedFailover"
      ]
      runbook_id = module.asr-dr-runbooks.automation-runbook-stop-apps-id
    }
 }   

  recovery_group {
    type = "Failover"
  }

  recovery_group {
    type                       = "Boot"
    replicated_protected_items = local.replication_protected_db_items

    post_action {
      name            = "db-health-check"
      type            = "AutomationRunbookActionDetails"
      fabric_location = "Primary"

      fail_over_directions = [
        "PrimaryToRecovery", "RecoveryToPrimary"
      ]
      fail_over_types = [
        "TestFailover", "PlannedFailover", "UnplannedFailover"
      ]
      runbook_id = module.asr-dr-runbooks.automation-runbook-db-health-check-id
    }
  }

  recovery_group {
    type = "Boot"
    replicated_protected_items = local.replication_protected_app_items

    post_action {
      name            = "test"
      type            = "AutomationRunbookActionDetails"
      fabric_location = "Primary"

      fail_over_directions = [
        "PrimaryToRecovery", "RecoveryToPrimary"
      ]
      fail_over_types = [
        "TestFailover", "PlannedFailover", "UnplannedFailover"
      ]
      runbook_id = module.asr-dr-runbooks.automation-runbook-test-id
    }
  }

  azure_to_azure_settings {
    primary_zone  = local.source_zone
    recovery_zone = local.target_zone
  }
}
```

### Debug Output/Panic Output

```shell
{"id":"/Subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationRecoveryPlans/example-recovery-plan","name":"example-recovery-plan","type":"Microsoft.RecoveryServices/vaults/replicationRecoveryPlans","properties":{"friendlyName":"example-recovery-plan","primaryFabricId":"/Subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationFabrics/primary-fabric","primaryFabricFriendlyName":"Switzerland North","recoveryFabricId":"/Subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationFabrics/primary-fabric","recoveryFabricFriendlyName":"Switzerland North","failoverDeploymentModel":"ResourceManager","replicationProviders":["A2A"],"allowedOperations":["UnplannedFailover","TestFailover"],"currentScenario":{"scenarioName":"UnplannedFailover","jobId":"/Subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationJobs/c272de86-fd6a-4032-93ed-ee3e8cbe5623","startTime":"2023-06-23T05:30:06.1067293Z"},"currentScenarioStatus":"Cancelled","currentScenarioStatusDescription":"Cancelled","groups":[{"groupType":"Shutdown","replicationProtectedItems":[],"startGroupActions":[{"actionName":"some-pre-action","failoverTypes":["PlannedFailover","UnplannedFailover","TestFailover"],"failoverDirections":["PrimaryToRecovery","RecoveryToPrimary"],"customDetails":{"instanceType":"AutomationRunbookActionDetails","runbookId":"/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/test/runbooks/some-runbook","fabricLocation":"Primary","canWrite":false,"canRead":true}},{"actionName":"stop-applications","failoverTypes":["PlannedFailover","UnplannedFailover","TestFailover"],"failoverDirections":["PrimaryToRecovery","RecoveryToPrimary"],"customDetails":{"instanceType":"AutomationRunbookActionDetails","runbookId":"/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/test/runbooks/Stop-Applications","fabricLocation":"Primary","canWrite":false,"canRead":true}}],"endGroupActions":[]},{"groupType":"Failover","replicationProtectedItems":[],"startGroupActions":[],"endGroupActions":[]},{"groupType":"Boot","replicationProtectedItems":[{"id":"/Subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationFabrics/primary-fabric/replicationProtectionContainers/primary-protection-container/replicationProtectedItems/zachadevtestpr01008","virtualMachineId":"055d5a72-a050-5823-bffb-1c264f4bfadb"}],"startGroupActions":[],"endGroupActions":[{"actionName":"db-health-check","failoverTypes":["PlannedFailover","UnplannedFailover","TestFailover"],"failoverDirections":["PrimaryToRecovery","RecoveryToPrimary"],"customDetails":{"instanceType":"AutomationRunbookActionDetails","runbookId":"/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/test/runbooks/DB-Health-Check","fabricLocation":"Primary","canWrite":false,"canRead":true}}]},{"groupType":"Boot","replicationProtectedItems":[],"startGroupActions":[],"endGroupActions":[{"actionName":"test","failoverTypes":["PlannedFailover","UnplannedFailover","TestFailover"],"failoverDirections":["PrimaryToRecovery","RecoveryToPrimary"],"customDetails":{"instanceType":"AutomationRunbookActionDetails","runbookId":"/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/test/runbooks/runbook-test","fabricLocation":"Primary","canWrite":false,"canRead":true}}]}],"providerSpecificDetails":[{"instanceType":"A2A","primaryZone":"1","recoveryZone":"2","primaryExtendedLocation":null,"recoveryExtendedLocation":null}]}}:"timestamp=2023-06-23T08":"48":32.128Z
2023-06-23T08:48:32.139Z [WARN]  Provider "registry.terraform.io/hashicorp/azurerm" produced an unexpected new value for azurerm_site_recovery_replication_recovery_plan.recovery_plan_az1 during refresh.
- .recovery_group: planned set element cty.ObjectVal(map[string]cty.Value{"post_action":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.NullVal(cty.String), "name":cty.StringVal("db-health-check"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/DB-Health-Check"), "script_path":cty.NullVal(cty.String), "type":cty.StringVal("AutomationRunbookActionDetails")})}), "pre_action":cty.ListValEmpty(cty.Object(map[string]cty.Type{"fabric_location":cty.String, "fail_over_directions":cty.Set(cty.String), "fail_over_types":cty.Set(cty.String), "manual_action_instruction":cty.String, "name":cty.String, "runbook_id":cty.String, "script_path":cty.String, "type":cty.String})), "replicated_protected_items":cty.ListVal([]cty.Value{cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationFabrics/primary-fabric/replicationProtectionContainers/primary-protection-container/replicationProtectedItems/zachadevtestpr01008")}), "type":cty.StringVal("Boot")}) does not correlate with any element in actual
- .recovery_group: planned set element cty.ObjectVal(map[string]cty.Value{"post_action":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.NullVal(cty.String), "name":cty.StringVal("restart-preprod-at-failback"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/Failback-PreProd-Restart"), "script_path":cty.NullVal(cty.String), "type":cty.StringVal("AutomationRunbookActionDetails")}), cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.NullVal(cty.String), "name":cty.StringVal("commit-recovery-plan"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/some-pre-action"), "script_path":cty.NullVal(cty.String), "type":cty.StringVal("AutomationRunbookActionDetails")})}), "pre_action":cty.ListValEmpty(cty.Object(map[string]cty.Type{"fabric_location":cty.String, "fail_over_directions":cty.Set(cty.String), "fail_over_types":cty.Set(cty.String), "manual_action_instruction":cty.String, "name":cty.String, "runbook_id":cty.String, "script_path":cty.String, "type":cty.String})), "replicated_protected_items":cty.ListValEmpty(cty.String), "type":cty.StringVal("Boot")}) does not correlate with any element in actual
- .recovery_group: planned set element cty.ObjectVal(map[string]cty.Value{"post_action":cty.ListValEmpty(cty.Object(map[string]cty.Type{"fabric_location":cty.String, "fail_over_directions":cty.Set(cty.String), "fail_over_types":cty.Set(cty.String), "manual_action_instruction":cty.String, "name":cty.String, "runbook_id":cty.String, "script_path":cty.String, "type":cty.String})), "pre_action":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.StringVal(""), "name":cty.StringVal("free-up-crg-at-failover"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/Free-Up-CRG"), "script_path":cty.StringVal(""), "type":cty.StringVal("AutomationRunbookActionDetails")}), cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.StringVal(""), "name":cty.StringVal("stop-applications"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/Stop-Applications"), "script_path":cty.StringVal(""), "type":cty.StringVal("AutomationRunbookActionDetails")}), cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.StringVal(""), "name":cty.StringVal("stop-dbs"), "runbook_id":cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Automation/automationAccounts/zacha9999pr01ns1asr01/runbooks/Stop-Databases"), "script_path":cty.StringVal(""), "type":cty.StringVal("AutomationRunbookActionDetails")}), cty.ObjectVal(map[string]cty.Value{"fabric_location":cty.StringVal("Primary"), "fail_over_directions":cty.SetVal([]cty.Value{cty.StringVal("PrimaryToRecovery"), cty.StringVal("RecoveryToPrimary")}), "fail_over_types":cty.SetVal([]cty.Value{cty.StringVal("PlannedFailover"), cty.StringVal("TestFailover"), cty.StringVal("UnplannedFailover")}), "manual_action_instruction":cty.StringVal("Please check if all scrips run successfully"), "name":cty.StringVal("approval-gate"), "runbook_id":cty.StringVal(""), "script_path":cty.StringVal(""), "type":cty.StringVal("ManualActionDetails")})}), "replicated_protected_items":cty.ListVal([]cty.Value{cty.StringVal("/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.RecoveryServices/vaults/zacha9999pr01ns1rsv90/replicationFabrics/primary-fabric/replicationProtectionContainers/primary-protection-container/replicationProtectedItems/zachadevtestpr01008")}), "type":cty.StringVal("Shutdown")}) does not correlate with any element in actual
2023-06-23T08:48:32.167Z [WARN]  Provider "registry.terraform.io/hashicorp/azurerm" produced an invalid plan for azurerm_site_recovery_replication_recovery_plan.recovery_plan_az1, but we are tolerating it because it is using the legacy plugin SDK.
The following problems may be the cause of any confusing errors from downstream operations:
- .azure_to_azure_settings[0].primary_edge_zone: planned value cty.StringVal("") for a non-computed attribute
- .azure_to_azure_settings[0].recovery_edge_zone: planned value cty.StringVal("") for a non-computed attribute
2023-06-23T08:48:32.306Z [DEBUG] provider.terraform-provider-azurerm_v3.61.0_x5: AzureRM Response for https://management.azure.com/subscriptions/xxxxxxx/resourceGroups/xxxxxxx/providers/Microsoft.Web/sites/overall-dr-function-app/config/slotConfigNames?api-version=2021-03-01:
HTTP/2.0 200 OK
Cache-Control: no-cache
Content-Type: application/json
Date: Fri, 23 Jun 2023 08:48:32 GMT
Expires: -1
Pragma: no-cache
Server: Microsoft-IIS/10.0
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Aspnet-Version: 4.0.30319
X-Content-Type-Options: nosniff
X-Ms-Correlation-Request-Id: 7369947b-3c6e-be1a-3760-978555942bb6
X-Ms-Ratelimit-Remaining-Subscription-Reads: 11965
X-Ms-Request-Id: 2a61caa7-d6f5-415d-961c-6c1d963043d2
X-Ms-Routing-Request-Id: SWITZERLANDWEST:20230623T084832Z:795a30e3-ba05-4e4b-8a70-9c5505fa7201
X-Powered-By: ASP.NET
```

### Expected Behaviour

`recovery_group` order should be the same as defined in terraform file to keep order on state validation 

### Actual Behaviour

`recovery_group`s will be listed as changed on each terraform plan regardless of actual changes 

### Steps to Reproduce

`terraform plan`

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #22258

Screenshot 2023-06-27 at 09 37 17

rerun the failed one
Screenshot 2023-06-27 at 09 37 34

### CODE REVIEW

File: website/docs/r/site_recovery_replication_recovery_plan.html.markdown [hunk 1]
```diff
@@ -258,7 +258,7 @@ A `recovery_group` block supports the following:
 
 * `type` - (Required) The Recovery Plan Group Type. Possible values are `Boot`, `Failover` and `Shutdown`.
 
-* `replicated_protected_items` - (Optional) (required) one or more id of protected VM.
+* `replicated_protected_items` - (Optional) (required) one or more id of protected VM. It must not be specified when `type` is `Shutdown`.
```
Comment (stephybun) [inline website/docs/r/site_recovery_replication_recovery_plan.html.markdown:261]: Is this property Optional or Required?

```suggestion
* `replicated_protected_items` - (Optional) (required) One or more protected VM IDs. It must not be specified when `type` is `Shutdown`.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Can we add a test to replicate the issue and validate that this indeed fixes it?

**Comment by ziyeqf** [comment]
As there is no code changes in resource in this commit, just run the specified test
---
```
❯ tftest recoveryservices TestAccSiteRecoveryReplicationRecoveryPlan_wrongSettings
=== RUN   TestAccSiteRecoveryReplicationRecoveryPlan_wrongSettings
=== PAUSE TestAccSiteRecoveryReplicationRecoveryPlan_wrongSettings
=== CONT  TestAccSiteRecoveryReplicationRecoveryPlan_wrongSettings
--- PASS: TestAccSiteRecoveryReplicationRecoveryPlan_wrongSettings (3786.00s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      3786.049s
```

**Comment by ziyeqf** [comment]
just updated code and triggered a test on TC, will update the result here.

**Comment by ziyeqf** [comment]
Screenshot 2023-06-29 at 11 08 46

**Review by stephybun — APPROVED** [review]
Thanks @ziyeqf LGTM 👍

**Pull request merged**